test/local-fs.spec.ts

```diff
@@ -17,14 +28,61 @@ describe('the local filesystem implementation', function () {
     after(()=>{
         dirCleanup();
     });
-    assertFileSystemContract(
-        () => {
-            const testPath = join(rootPath, 'fs_'+(counter++));
-            mkdirSync(testPath);
-            return new LocalFileSystem(testPath).init();
-        }, {
-            interval: 50,
-            noExtraEventsGrace: 150,
-            timeout: 1500
+
+    function getFS() {
+        testPath = join(rootPath, 'fs_'+(counter++));
+        return mkdirAsync(testPath).then(() => new LocalFileSystem(testPath).init());
+    }
+
+    const eventMatcherOptions: EventsMatcher.Options = {
+        interval: 50,
+        noExtraEventsGrace: 150,
+        timeout: 1500
+    };
+
+    assertFileSystemContract(getFS, eventMatcherOptions);
+
+    describe(`external changes`, () => {
+        const fileName = 'foo.txt';
+        const dirName = 'dir';
+        const content = 'content';
+
+        let fs: FileSystem;
+        let matcher: EventsMatcher;
+        beforeEach(() => {
+            matcher = new EventsMatcher(eventMatcherOptions)
+            return getFS().then(newFs => {
+                fs = newFs
+                matcher.track(fs.events as any as EventEmitter,
+                    'fileCreated', 'fileChanged', 'fileDeleted', 'directoryCreated', 'directoryDeleted');
+            })
         });
+
+        it(`handle dir creation and deletion`, () => {
+            return mkdirAsync(`${testPath}/${dirName}`)
+                .then(() => waitFor(
+                    () => expect(fs.loadDirectoryTree())
```

Comment:
also, what happens when `fs.loadDIrectoryTree()` is never resolved?

the `waitFor` driver should have some tests of its own.